qiskit/opflow/gradients/circuit_gradients/lin_comb.py

```diff
@@ -596,36 +606,38 @@ def apply_grad_gate(
     def _gradient_states(
         self,
         state_op: StateFn,
-        meas_op: Union[OperatorBase, bool] = True,
+        meas_op: Optional[OperatorBase] = None,
         target_params: Optional[Union[Parameter, List[Parameter]]] = None,
         open_ctrl: bool = False,
         trim_after_grad_gate: bool = False,
+        aux_meas_op: OperatorBase = Z,
     ) -> ListOp:
         """Generate the gradient states.
-
         Args:
             state_op: The operator representing the quantum state for which we compute the gradient.
             meas_op: The operator representing the observable for which we compute the gradient.
             target_params: The parameters we are taking the gradient wrt: ω
-            open_ctrl: If True use an open control for ``grad_gate`` instead of closed.
-            trim_after_grad_gate: If True remove all gates after the ``grad_gate``. Can
-                be used to reduce the circuit depth in e.g. computing an overlap of gradients.
-
+            aux_meas_op: The operator that the auxiliary qubit is measured with respect to.
+                 for aux_meas_op = Z we compute Re[(dω⟨ψ(ω)|)O(θ)|ψ(ω)〉]
+                 for aux_meas_op = -Y we compute Im[(dω⟨ψ(ω)|)O(θ)|ψ(ω)〉]
         Returns:
             ListOp of StateFns as quantum circuits which are the states w.r.t. which we compute the
             gradient. If a parameter appears multiple times, one circuit is created per
             parameterized gates to compute the product rule.
-
         Raises:
             AquaError: If one of the circuits could not be constructed.
             TypeError: If the operators is of unsupported type.
         """
-        qr_superpos = QuantumRegister(1)
-        state_qc = QuantumCircuit(*state_op.primitive.qregs, qr_superpos)
-        state_qc.h(qr_superpos)
         # unroll separately from the H gate since we need the H gate to be the first
         # operation in the data attributes of the circuit
         unrolled = self._transpile_to_supported_operations(state_op.primitive, self.SUPPORTED_GATES)
+        qr_superpos = QuantumRegister(1)
+        state_qc = QuantumCircuit(*state_op.primitive.qregs, qr_superpos)
+        state_qc.h(qr_superpos)
+
+        if meas_op is not None:
+            meas_op = meas_op.reduce()
```

Comment:
Is this required? Can `Z` or `-Y` be further reduced? 🤔

Reply:
This is again a safety move to ensure that things are put together right. But the unit tests succeed without it. 
If there is a strong opinion for removing it -- we can do that.